### PR TITLE
adding line tool

### PR DESCRIPTION
**Pull request by 1ntEgr8**
line tool (beta)

### CODE REVIEW

File: js/tools.js [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+ */
+class Canvas {
+    constructor(canvasId, canvasContainerId) {
+        // svg canvas
+        this.canvas = document.getElementById(canvasId);  
+        // container for the canvas to support accurate mouse click coordinates
+        this.canvasContainer = document.getElementById(canvasContainerId);
+        // list of shapes drawn on the canvas; shapes must be of class Shape
+        this.shapes = [];  
+        this.currentShape = null;
```
Comment (joel99) [inline js/tools.js]: not too supportive for multiple shapes but i'm ok w/ that for now

File: js/tools.js [hunk 2]
```diff
@@ -0,0 +1,334 @@
+let selectButton = document.getElementById("select");
```
Comment (joel99) [inline js/tools.js]: I think these and some considerable bit below should be wrapped in an `onload`

File: js/tools.js [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+            path.setAttributeNS(null, attribute, this.attributes[attribute]);
+        }
+        pathGroup.appendChild(path);
+        pathGroup.classList.add("path");
+        return pathGroup;
+    }
+
+    showNodes() {
+        for (let i = 0; i < this.nodes.length; i++) {
+            this.nodes[i].domElement.style.visibility = 'visible';
```
Comment (joel99) [inline js/tools.js]: I would add style controlling methods to `Node`

File: js/tools.js [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+
+/**
+ * This class represents a Shape drawn on the Canvas
+ */
+class Shape {
+    constructor(svgTag, attributes) {
+        this.svgTag = svgTag;
+        this.attributes = attributes;
+        this.domGroup = this.create();
+        this.nodes = [];
```
Comment (joel99) [inline js/tools.js]: should `nodes` be an attribute at this level if `Node` is a subclass?

File: js/tools.js (hunk 1 above)
Comment (1ntEgr8) [reply js/tools.js]: the shapes array is being used for selection purposes. Without it, it becomes very difficult to track down which shape has been selected, and also to which shape styles need to be addded

File: js/tools.js (hunk 2 above)
Comment (1ntEgr8) [reply js/tools.js]: fixing this will require some amount of refactoring; creating a separate issue for this

**Pull request merged**